### PR TITLE
Unit testing: Whitelisting test names per backend type

### PR DESCRIPTION
**Pull request by zhztheplayer**
Move whitelist from UT classes to BackendTestSettings(renamed from NotSupport since it currently manages both inclusion/exclusion). So that CH be and Velox be don't affect the other when adding whitelist for one of them.

### CODE REVIEW

File: gluten-ut/src/test/scala/org/apache/spark/sql/GlutenTestsBaseTrait.scala [hunk 1]
```diff
@@ -27,40 +28,29 @@ trait GlutenTestsBaseTrait {
...
-        val issueIDPattern = "SPARK-[0-9]+".r
-        val issueID = issueIDPattern.findFirstIn(testName) match {
-          case Some(x: String) => x
-        }
-        fuzzyContain = blackTestNameList.exists(_.startsWith(issueID))
+  def shouldRun(testName: String): Boolean = {
+    if (testName.startsWith(GLUTEN_TEST)) {
+      return true
+    }
+    if (testNameBlackList.exists(_.equalsIgnoreCase(GlutenTestConstants.IGNORE_ALL))) {
```
Comment (loneylee) [inline gluten-ut/src/test/scala/org/apache/spark/sql/GlutenTestsBaseTrait.scala:38]: Is exclude case should add to testNameBlackList

Comment (zhztheplayer) [reply gluten-ut/src/test/scala/org/apache/spark/sql/GlutenTestsBaseTrait.scala:39]: No need to do that. And I added following comment to `testNameblackList`

https://github.com/oap-project/gluten/blob/558f35a6e4e82eda9d074384f512584a7b767d6c/gluten-ut/src/test/scala/org/apache/spark/sql/GlutenTestsBaseTrait.scala#L31-L33

Does the comment look clear enough to you?

**Review by lviiii — APPROVED** [review]
LGTM

**Comment by zhztheplayer** [comment]
I am merging the patch and please ping me if any issue is found. Thanks guys!

**Pull request merged**